chia/full_node/mempool_manager.py

```diff
@@ -131,7 +133,9 @@ def shut_down(self) -> None:
         self.pool.shutdown(wait=True)
 
     async def create_bundle_from_mempool(
-        self, last_tb_header_hash: bytes32
+        self,
+        last_tb_header_hash: bytes32,
+        item_inclusion_filter: Optional[Callable[[MempoolManager, MempoolItem], bool]] = None,
```

Comment:
this looks unrelated to preserving the block height when rebuilding mempool. in an ideal world, it would be a separate PR or at least commit. This is necessary because the coin-selection logic has a filter, right? Or is this a debug-only feature?

Reply:
This is essentially a debug-only feature. I was hoping to leave this out of this particular PR, because the testing support code is much larger than the actual fix. But I do next it for 2 other PRs, and this test would also be very difficult without it.

Here's why we need the filter: the MempoolManager algorithm to select spends for a block is not modular.
Also, the code to create new MempoolItem instances is deep inside `validate_spend_bundle`, which is several calls down from the main action.

What needs to happen eventually, is we need to refactor some of the giant methods in MempoolManager, but definitely not now - this is as small as I can make it and still test the production code.

At first, I was reluctant to include this is main - I didn't want to have people messing around with the Mempool selection algorithm. But on further reflection, that only matters for consensus.